src/index.mjs

```diff
@@ -113,10 +113,17 @@ export async function main(req, ctx) {
     if (req.method === 'GET' && new URL(req.url).pathname.startsWith('/info.json')) {
       return respondInfo(ctx);
     }
+    const isDirList = (new URL(req.url).pathname.endsWith('/'));
```

Comment:
We're doing a lot of `new URL(req.url).pathname` in this method. It might be an idea to put it in a local `const`.

Reply:
ack